src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -365,7 +365,7 @@ private void SolutionEvents_AfterClosing()
                             }
 
                             // Upgrade request if necessary
-                            if (next.RestoreSource != request.RestoreSource)
+                            if (next?.RestoreSource != request.RestoreSource)
```

Comment:
Actually the right thing to do would be
```cs
if (next != null && next.RestoreSource != request.RestoreSource)
{
...
}
```
Otherwise you'll have the same problem at line 373 below.